lib/ansible/modules/storage/hpe/hpe3par_cpg.py

```diff
@@ -0,0 +1,435 @@
+        growth_increment="{{ growth_increment }}"
+        growth_increment_unit="{{ growth_increment_unit }}"
+        growth_limit="{{ growth_limit }}"
+        growth_limit_unit="{{ growth_limit_unit }}"
+        growth_warning="{{ growth_warning }}"
+        growth_warning_unit="{{ growth_warning_unit }}"
+        raid_type="{{ raid_type }}"
+        set_size="{{ set_size }}"
+        high_availability="{{ high_availability }}"
+        disk_type="{{ disk_type }}"
+
+    - name: Delete CPG "{{ cpg_name }}"
+      hpe3par_cpg:
+        storage_system_ip="{{ storage_system_ip }}"
+        storage_system_username="{{ storage_system_username }}"
+        storage_system_password="{{ storage_system_password }}"
+        state=absent
+        cpg_name="{{ cpg_name }}"
+'''
+
+RETURN = r'''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from hpe3par_sdk import client
+except ImportError:
+    client = None
+
+
+def convert_to_binary_multiple(size, size_unit):
+    size_mib = 0
+    if size_unit == 'GiB':
+        size_mib = size * 1024
+    elif size_unit == 'TiB':
+        size_mib = size * 1048576
+    elif size_unit == 'MiB':
+        size_mib = size
+    return int(size_mib)
+
+
+def validate_set_size(raid_type, set_size):
+    if raid_type is not None or set_size is not None:
+        set_size_array = client.HPE3ParClient.RAID_MAP[raid_type]['set_sizes']
+        if set_size in set_size_array:
+            return True
+    return False
+
+
+def cpg_ldlayout_map(ldlayout_dict):
+    if ldlayout_dict['RAIDType'] is not None and ldlayout_dict['RAIDType']:
+        ldlayout_dict['RAIDType'] = client.HPE3ParClient.RAID_MAP[
+            ldlayout_dict['RAIDType']]['raid_value']
+    if ldlayout_dict['HA'] is not None and ldlayout_dict['HA']:
+        ldlayout_dict['HA'] = getattr(
+            client.HPE3ParClient, ldlayout_dict['HA'])
+    return ldlayout_dict
+
+
+def create_cpg(
+        client_obj,
+        storage_system_username,
+        storage_system_password,
+        cpg_name,
+        domain,
+        growth_increment,
+        growth_increment_unit,
+        growth_limit,
+        growth_limit_unit,
+        growth_warning,
+        growth_warning_unit,
+        raid_type,
+        set_size,
+        high_availability,
+        disk_type):
+    if storage_system_username is None or storage_system_password is None:
+        return (
+            False,
+            False,
+            "CPG create failed. Storage system username or password is null",
+            {})
+    if cpg_name is None:
+        return (False, False, "CPG create failed. CPG name is null", {})
+    if len(cpg_name) < 1 or len(cpg_name) > 31:
+        return (False, False, "CPG create failed. CPG name must be atleast 1 character and not more than 31 characters", {})
+    if not validate_set_size(raid_type, set_size):
+        return (False, False, "Set size not part of RAID set", {})
+    try:
+        validate_set_size(raid_type, set_size)
```

Comment:
This 2nd call to `validate_set_size` must be removed.

Reply:
Removed